gcloud-java-storage/src/test/java/com/google/gcloud/storage/ITStorageTest.java

```diff
@@ -334,30 +332,32 @@ public void testCopyBlob() {
   @Test
   public void testCopyBlobUpdateMetadata() {
     String sourceBlobName = "test-copy-blob-update-metadata-source";
-    BlobInfo sourceBlob = BlobInfo.builder(BUCKET, sourceBlobName).build();
-    assertNotNull(storage.create(sourceBlob));
-    String targetBlobName = "test-copy-blob-update-metadata-target";
-    BlobInfo targetBlob =
-        BlobInfo.builder(BUCKET, targetBlobName).contentType(CONTENT_TYPE).build();
-    Storage.CopyRequest req = Storage.CopyRequest.of(BUCKET, sourceBlobName, targetBlob);
-    BlobInfo remoteBlob = storage.copy(req);
-    assertNotNull(remoteBlob);
-    assertEquals(targetBlob.blobId(), remoteBlob.blobId());
-    assertEquals(CONTENT_TYPE, remoteBlob.contentType());
+    BlobId source = BlobId.of(BUCKET, sourceBlobName);
+    assertNotNull(storage.create(BlobInfo.builder(source).build(), BLOB_BYTE_CONTENT));
+    String targetBlobName = "test-copy-blob-target";
+    BlobInfo target = BlobInfo.builder(BUCKET, targetBlobName)
+        .contentType(CONTENT_TYPE)
+        .metadata(ImmutableMap.of("k", "v"))
+        .build();
+    Storage.CopyRequest req = Storage.CopyRequest.of(source, target);
+    CopyWriter copyWriter = storage.copy(req);
+    assertEquals(copyWriter.result(), storage.get(BUCKET, targetBlobName));
```

Comment:
Shall we also verify the content-type and metadata that they are what we expect them to be?